### PR TITLE
Imgzz

### PR DESCRIPTION
**Pull request by Hellowlol**

### CODE REVIEW

**Comment by JonnyWong16** [comment]
```
2016-05-08 22:44:01 - ERROR   :: CP Server Thread-12 : Failed to delete F:\Users\Jonny\Documents\GitHub\plexpy\cache\images [Error 145] The directory is not empty: 'F:\\Users\\Jonny\\Documents\\GitHub\\plexpy\\cache\\images'
Traceback (most recent call last):
  File "F:\Users\Jonny\Documents\GitHub\plexpy\plexpy\webserve.py", line 2892, in delete_image_cache
    os.rmdir(cache_dir)
WindowsError: [Error 145] The directory is not empty: 'F:\\Users\\Jonny\\Documents\\GitHub\\plexpy\\cache\\images'
```

Can trackbacks be formatted?
image

---

Image loading is noticeably faster for me (or maybe just placebo effect). :stuck_out_tongue:

**Comment by Hellowlol** [comment]
Formatted how? the old version simply ignored the traceback.. :P. Ill fix the delete function.

Did you you think about newlines etc or using something like < pre >

File: plexpy/webserve.py [hunk 1]
```diff
@@ -1902,11 +1905,17 @@ def get_sync(self, machine_id=None, user_id=None, **kwargs):
 
 
     ##### Logs #####
+    @cherrypy.expose
```
Comment (Hellowlol) [inline plexpy/webserve.py:1909]: oops. remove this.

**Comment by JonnyWong16** [comment]
Formatted with line breaks. Will this work? L1954

```
filt[tl][2] += '\n' + l.replace('\n', '')
```

**Comment by Hellowlol** [comment]
@JonnyWong16  What about something like this? image

Only the tracebacks gets a pre tag.

**Comment by JonnyWong16** [comment]
That will work.

What about this?

```
# Add traceback message to previous msg..
tl = (len(filt) - 1)
m = re.match(r'\s+', l)
n = m.end() if m else 0
l = '&nbsp;' * (2*n) + l[n:]
filt[tl][2] += '<br>' + l
```

Without regex

```
# Add traceback message to previous msg..
tl = (len(filt) - 1)
n = len(l) - len(l.lstrip(' '))
l = '&nbsp;' * (2*n) + l[n:]
filt[tl][2] += '<br>' + l
```

image

**Comment by JonnyWong16** [comment]
Deleting image cache works now.

**Comment by Hellowlol** [comment]
I used the pre tag so the message should be stand out, but we can do something like that if you want to. I still want to a void the regex since regex are slow. My old regex solution was 10 times slower then the lame function

**Comment by JonnyWong16** [comment]
It stands out a bit too much lol. But I can always play around with the css.

**Comment by Hellowlol** [comment]
Ok, should be good to go it you want to fix it with css if not it let me know and I'll do your non regex way

**Comment by JonnyWong16** [comment]
I'll clean it up. Thanks.

**Comment by JonnyWong16** [comment]
@Hellowlol Not sure if it's just my permission issues

```
2016-05-12 21:48:33 - ERROR :: CP Server Thread-6 : Failed to create F:\Users\Jonny\Documents\GitHub\plexpy\cache\images: [Error 5] Access is denied: 'F:\\Users\\Jonny\\Documents\\GitHub\\plexpy\\cache\\images'.
Traceback (most recent call last):
  File "F:\Users\Jonny\Documents\GitHub\plexpy\plexpy\webserve.py", line 2945, in delete_cache
    os.makedirs(cache_dir)
  File "C:\Python27\lib\os.py", line 157, in makedirs
    mkdir(name, mode)
WindowsError: [Error 5] Access is denied: 'F:\\Users\\Jonny\\Documents\\GitHub\\plexpy\\cache\\images'
```

**Comment by Hellowlol** [comment]
Must be. I have never seen it on windows before :P

**Pull request merged**